### PR TITLE
Custom loader & fix for #7

### PR DESCRIPTION
**Pull request by SimonB407**

### CODE REVIEW

File: example-app/package.json [hunk 1]
```diff
@@ -18,14 +18,15 @@
     "test:e2e": "jest --config ./test/jest-e2e.json"
   },
   "dependencies": {
-    "@admin-bro/express": "^3.0.0-beta.3",
-    "@admin-bro/mongoose": "^1.0.0-beta.3",
-    "@admin-bro/nestjs": "^1.0.0-beta.2",
```
Comment (plorencrstit) [inline example-app/package.json:23]: Can you remove it for sure?

File: src/loaders/express.loader.ts [hunk 2]
```diff
@@ -23,43 +23,73 @@ export class ExpressLoader extends AbstractLoader {
...
+
+    app.use(options.adminBroOptions.rootPath, router);
+    this.reorderRoutes(app, options);980
+  }
+
+  private reorderRoutes(app, options) {
     let jsonParser;
     let urlencodedParser;
+    let admin;
+
```
Comment (plorencrstit) [inline src/loaders/express.loader.ts]: remove blank line

**Review by plorencrstit — APPROVED** [review]

File: example-app/package.json (hunk 1 above)
Comment (SimonB407) [reply example-app/package.json:23]: Yes, it was used in previous version in example-app, but it's silly to link lib from npm for example app that is created for that exact lib.

File: src/loaders/express.loader.ts (hunk 2 above)
Comment (SimonB407) [reply src/loaders/express.loader.ts]: Done

File: src/loaders/express.loader.ts [hunk 3]
```diff
@@ -23,43 +23,72 @@ export class ExpressLoader extends AbstractLoader {
...
+      loadPackage('express-session', '@admin-bro/nestjs');
+      router = adminBroExpressjs.buildAuthenticatedRouter(
+        admin, options.auth, undefined, options.sessionOptions, options.formidableOptions
+      );
+    } else {
+      router = adminBroExpressjs.buildRouter(admin, undefined, options.formidableOptions);
+    }
+
+    app.use(options.adminBroOptions.rootPath, router);
+    this.reorderRoutes(app, options);980
```
Comment (JonaszJestem) [inline src/loaders/express.loader.ts:38]:
```suggestion
    this.reorderRoutes(app, options);
```

File: src/loaders/express.loader.ts [hunk 4]
```diff
@@ -23,43 +23,72 @@ export class ExpressLoader extends AbstractLoader {
...
-    }
+      // if admin-bro-nestjs didn't reorder the middleware
+      // the body parser would have come after corsMiddleware
+      const corsIndex = app._router.stack.findIndex(
+        (layer: { name: string }) => layer.name === 'corsMiddleware',
+      )
 
-    app.use(options.adminBroOptions.rootPath, router);
+      // in other case if there is no corsIndex we go after expressInit, because right after that
+      // there are nest endopoints.
```
Comment (JonaszJestem) [inline src/loaders/express.loader.ts:78]:
```suggestion
      // there are nest endpoints.
```

File: example-app/package.json [hunk 5]
```diff
@@ -18,14 +18,15 @@
...
+    "@admin-bro/express": "^3.0.1",
+    "@admin-bro/mongoose": "^1.1.0",
     "@nestjs/common": "^6.7.2",
     "@nestjs/core": "^6.7.2",
     "@nestjs/mongoose": "^7.0.2",
     "@nestjs/platform-express": "^6.7.2",
-    "admin-bro": "^3.0.0-beta.12",
+    "admin-bro": "^3.3.1",
```
Comment (JonaszJestem) [inline example-app/package.json:27]: shouldn't all of these dependencies be as `peerDependencies` and `devDependencies`?
I think that `admin-bro-nestjs` shouldn't really install any dependencies for itself as every user would install them in their projects

**Review by JonaszJestem — APPROVED** [review]

File: src/loaders/express.loader.ts (hunk 3 above)
Comment (SimonB407) [reply src/loaders/express.loader.ts:38]: God damn it :laughing: My cat wanted to add something too :cat:

File: example-app/package.json (hunk 5 above)
Comment (SimonB407) [reply example-app/package.json:27]: It's for example-app. Honestly I am not sure if this package.json is even resolved when installing library.

**Pull request merged**